joomla/joomla-cms
# Improve plugins scaffolding for custom fields

#### This is a redo of https://github.com/joomla/joomla-cms/pull/4294
#### Summary of Changes

Right now if you want to include some custom fields in your plugin you need to include a custom field path in the relative xml file, something like:

```
<fieldset addfieldpath="/plugins/system/<plugin name>/field">
```

Documented here: https://docs.joomla.org/Creating_a_custom_form_field_type

This PR leverages this, so custom fields will happily reside inside a folder named `field`

Why field and not fields?
Singular word help us transition to autoload and namespace
#### Testing Instructions

Apply this patch with patch tester
Go to plugins tinyMCE and if you can see a dropdown for the skins you have a successful test:
![screen shot 2016-04-08 at 00 00 21](https://cloud.githubusercontent.com/assets/3889375/14366741/f9ee9ab2-fd1c-11e5-8dec-0003b46000b6.png)

Also go to Profile and check the Term of service and Date of birth, if dropdown then success:
![screen shot 2016-04-08 at 00 00 05](https://cloud.githubusercontent.com/assets/3889375/14366796/22da95fc-fd1d-11e5-80f3-772631fdede0.png)

Check the stats plugin and observe that all fields are correctly rendered/functioning
#### B/C

No backwards compatibility break
#### Performance

We just add one more folder check to the whole stack, so performance degradation should be unnoticeable. (Joomla itself does that hundreds of times for core libraries)
